08-graphs/01-graphs.md

```diff
@@ -18,41 +18,80 @@ By the end of this lesson we should be able to:
+
+Many, many problems can be represented by graphs even those not involving geographic or navigational problems.  Studying graph problems gives us another way to solve a broad sector of difficult problems.
+
+### !callout-info
+
+## Binary Search Trees, Linked Lists, and Graphs
 
-## Graph Description
+Binary search trees and linked lists are actually just subtypes of graphs! They are both collections of nodes with edges between them. There are just more specifications about the number and direction of edges allowed to be connected to a single node than there are with a general graph data strucutre.
 
-A graph is a data structure in computer science, consisting of a finite collection of _nodes_, also called _verticies_, and a collection of connections between nodes known as _edges_.  A graph can be either a _directed graph_, or an _undirected graph_.  In a directed graph, edges are not bidirectional.  So in the example below you travel from node A to node B, but not from node B to node A.
+### !end-callout
+
+### Directed vs Undirected Graphs
+
+Graphs can be _directed_ or _undirected_.
 
 **Directed Graph**
 
+In a directed graph, each edge is unidirectional. For any two nodes A and B, an edge from node A to B does not imply that there is an edge going from node B to node A.
+
+As depicted in the example graph below, we use arrows to depict the edges of directed graphs.
+
 ![Example Directed Graph](images/directed-graph.png)
 
-On the other hand, in an undirected graph, each edge is bidirectional.  In other words in the example graph below you can travel from Ada to Babbage and from Babbage to Ada.
+Directed graphs can be useful to represent one-way relationships. For example, imagine the graph below depicts available flights for an airline. We can see that there is a flight available from Mexico City to Lima, but there is no available flight from Lima back to Mexico City. Likewise, there is a flight from Paris to Istanbul, but no flight from Istanbul to Paris.
+
+Directed graphs can still show bidirectional relationships. For example, the graph also shows that there is a flight from Mexico City to Paris and back from Paris to Mexico City. 
+
+![Example Directed Graph Airline Flights](images/directed-graph-flights.png)
 
 **Undirected Graph**
 
+On the other hand, in an undirected graph, every edge is bidirectional. In other words, if we can travel from node A to node B, we can assume that we can also travel from node B to node A. 
+
+Undirected graphs use lines without arrows to depict edges between nodes. In the example graph below, the connection between Ada shows that we can travel from Ada to Babbage and from Babbage back to Ada.
+
 ![Example Undirected Graph](images/undirected-graph.png)
```

Comment:
I'm not sure if anyone will find this confusing or misleading, but I wanted to mention that this image shows two connections between Ada and Turing.